pkg/asset/agent/agentconfig/agent_config.go

```diff
@@ -34,8 +38,70 @@ func (*AgentConfig) Dependencies() []asset.Asset {
 	return []asset.Asset{}
 }
 
-// Generate generates the Agent manifest.
+// Generate generates the Agent Config manifest.
 func (a *AgentConfig) Generate(dependencies asset.Parents) error {
+
+	// TODO: We are temporarily generating a template of the agent-config.yaml
+	// Change this when its interactive survey is implemented.
+	agentConfigTemplate := `#
+# Note: This is an sample AgentConfig file showing
+# which fields are available to aid you in creating your 
+# own agent-config.yaml file.
+#
+apiVersion: v1
+kind: AgentConfig
+metadata:
+  name: example-agent-config
+  namespace: cluster0
+spec:
+# All fields are optional
+  rendezvousIP: your-node0-ip
+  hosts:
+  # If a host is listed, then at least one interface
+  # needs to be specified.
+  - hostname: change-to-hostname
+    role: master
+    # For more information about rootDeviceHints:
+    # https://docs.openshift.com/container-platform/4.10/installing/installing_bare_metal_ipi/ipi-install-installation-workflow.html#root-device-hints_ipi-install-installation-workflow
+    rootDeviceHints:
+	  deviceName: /dev/sda
+    interfaces:
+      - macAddress: 00:00:00:00:00:00
+        name: host-network-interface-name
+    # networkConfig contains the NMStateConfig.Spec.Config.interfaces definition for this host.
+    # https://nmstate.io/examples.html
+    # NMStateConfig.Spec.Interfaces section is not included in networkConfig. It is specified
+    # in the above interfaces: node
+    networkConfig:
+      interfaces:
+        - name: eth0
+          type: ethernet
+          state: up
+          mac-address: 00:00:00:00:00:00
+          ipv4:
+            enabled: true
+            address:
+              - ip: 192.168.122.2
+                prefix-length: 23
+            dhcp: false
+`
+
+	a.Template = agentConfigTemplate
```

Comment:
What happens if the user skips the `agent create agent-config-template` step and goes straight to `agent create image`? Won't this mean that we'll get the template passed to other assets as if it were real?

Reply:
Right now if the user doesn't create the agent-config template, then they see a message saying the nmstate config is missing:

Reply:
Ah, the improved error message will come from https://github.com/openshift/installer/pull/6099